pkg/webhooks/kafkacluster_validator.go

```diff
@@ -133,3 +160,67 @@ func getMissingMounthPathLocation(mounthPath string, kafkaClusterSpec *banzaiclo
 	}
 	return fromConfigGroup
 }
+
+// checkListeners validates the spec.listenersConfig object
+func checkListeners(kafkaClusterSpec *banzaicloudv1beta1.KafkaClusterSpec) field.ErrorList {
+	var allErrs field.ErrorList
+
+	// check values of containerPort
+	allErrs = append(allErrs, checkUniqueListenerContainerPort(kafkaClusterSpec.ListenersConfig)...)
+
+	// check values of externalStartingPort
+	allErrs = append(allErrs, checkExternalListenerStartingPort(kafkaClusterSpec)...)
+
+	return allErrs
+}
+
+// checkUniqueListenerContainerPort checks for duplicate containerPort numbers across both internal and external listeners
+// which would subsequently generate a "Duplicate value" error when creating a Service which accumulates all these ports.
+// The first time a port number is found will not be reported as duplicate; only subsequent instances using that port are.
+// (this is done to keep in tune with the way the K8s Service would report the "Duplicate value" error which ignores the first instance)
+func checkUniqueListenerContainerPort(listeners banzaicloudv1beta1.ListenersConfig) field.ErrorList {
+	var allErrs field.ErrorList
+	var containerPorts = make(map[int32]int)
+
+	for i, intLstnr := range listeners.InternalListeners {
+		containerPorts[intLstnr.ContainerPort] += 1
+		if containerPorts[intLstnr.ContainerPort] > 1 {
+			fldErr := field.Duplicate(field.NewPath("spec").Child("listenersConfig").Child("internalListeners").Index(i).Child("containerPort"), intLstnr.ContainerPort)
+			allErrs = append(allErrs, fldErr)
+		}
+	}
+	for i, extLstnr := range listeners.ExternalListeners {
+		containerPorts[extLstnr.ContainerPort] += 1
+		if containerPorts[extLstnr.ContainerPort] > 1 {
+			fldErr := field.Duplicate(field.NewPath("spec").Child("listenersConfig").Child("externalListeners").Index(i).Child("containerPort"), extLstnr.ContainerPort)
+			allErrs = append(allErrs, fldErr)
+		}
+	}
+
+	return allErrs
+}
+
+// checkExternalListenerStartingPort checks the generic sanity of the resulting external port (valid number between 1 and 65535)
+func checkExternalListenerStartingPort(kafkaClusterSpec *banzaicloudv1beta1.KafkaClusterSpec) field.ErrorList {
+	// if there are no externalListeners, there is no need to perform the rest of the checks in this function
+	if kafkaClusterSpec.ListenersConfig.ExternalListeners == nil {
+		return nil
+	}
+
+	var allErrs field.ErrorList
+	const maxPort int32 = 65535
+	for i, extLstn := range kafkaClusterSpec.ListenersConfig.ExternalListeners {
```

Comment:
Suggestion: `extLstn` is a bit ambitious to me
```suggestion
	for i, extListener := range kafkaClusterSpec.ListenersConfig.ExternalListeners {
```
But it's fine if you feel like keeping it like this since it is just used within the for loop